src/Sarif.FunctionalTests/v2/ConverterTestData/PREfast/Expected-022.xml.sarif

```diff
@@ -12,6 +12,53 @@
           "mimeType": "text/x-cpp"
         }
       },
+      "logicalLocations": {
+        "f": {
+          "name": "f",
+          "decoratedName": "?f@@YAXXZ"
+        },
+        "h": {
+          "name": "h",
+          "decoratedName": "?h@@YAXPAHH@Z"
+        },
+        "foo": {
+          "name": "foo",
+          "decoratedName": "?foo@@YAXPAHI@Z"
+        },
+        "StrLen": {
+          "name": "StrLen",
+          "decoratedName": "?StrLen@@YAIPBD@Z"
+        },
+        "BugLast": {
+          "name": "BugLast",
+          "decoratedName": "?BugLast@@YADPBD@Z"
+        },
+        "g1": {
+          "name": "g1",
+          "decoratedName": "?g1@@YAXPAH@Z"
+        },
+        "foo-0": {
+          "name": "foo",
+          "fullyQualifiedName": "foo",
+          "decoratedName": "?foo@@YADXZ"
+        },
+        "bar": {
+          "name": "bar",
+          "decoratedName": "?bar@@YAXPAHII@Z"
+        },
+        "ShiftTest::f": {
+          "name": "ShiftTest::f",
```

Comment:
Bug: `name` should be only the last component, `f`. That's what distinguishes it from the "fully qualified name" (`ShiftTest::f`).

Reply:
BTW, this might be a bug in the PREfast converter. If so, file it and move on.

Reply:
Yeah I haven't touched this

Reply:
Thanks.

@michaelcfanning  FYI there's a bug in the PREfast converter that populates `logicalLocation.name` with the FQLN, as in this example. @EasyRhin0 will file it.

Reply:
Actually I did add this to persist the decoratedName, so I'll fix.

Reply:
In that case... how sure are we that the other issue, where we populate `name` unnecessarily, is a converter issue? Could that also be your bug?

Reply:
Yeah same, though other converters have the same problem. FxCop:
        "wpfapplication5.exe": {
          "name": "wpfapplication5.exe",
          "fullyQualifiedName": "wpfapplication5.exe",
          "kind": "module"
        },

In the PREfast case, the Defect.Function property is used and it is the full string. Should I be parsing based on "::"?

Reply:
Ah never mind I think I have it.

Reply:
(Yes, split on `::`.)

Reply:
@EasyRhin0 @michaelcfanning File a separate bug for any converter with this non-optimal behavior.